--- sanlock-2.4/init.d/sanlock.orig	2012-07-23 16:24:38.000000000 +0200
+++ sanlock-2.4/init.d/sanlock	2012-09-02 21:09:14.181273151 +0200
@@ -32,94 +32,85 @@ SANLOCKOPTS="-U $SANLOCKUSER -G $SANLOCK
 [ -f /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
 
 start() {
-	[ -x $exec ] || exit 5
-
-	if [ ! -d /var/run/$prog ]; then
-		install -d -o $SANLOCKUSER -g $SANLOCKUSER -m 775 /var/run/$prog
-		[ -x /sbin/restorecon ] && restorecon /var/run/$prog
+	# Check if the service is already running?
+	if [ -f /var/lock/subsys/sanlock ]; then
+		msg_already_running "sanlock"
+		return
 	fi
 
-	echo -n $"Starting $prog: "
+	msg_starting "sanlock"
 	daemon $prog daemon $SANLOCKOPTS
-	retval=$?
-	echo
-	[ $retval -eq 0 ] && touch $lockfile
-	return $retval
+	RETVAL=$?
+	[ $RETVAL -eq 0 ] && touch $lockfile
+	return $RETVAL
 }
 
 stop() {
-	echo -n $"Sending stop signal $prog: "
+	if [ ! -f /var/lock/subsys/sanlock ]; then
+		msg_not_running "sanlock"
+		return
+	fi
 	killproc -p $runfile $prog -TERM
-	retval=$?
-	echo
+	RETVAL=$?
 
-	if [ $retval -ne 0 ]; then
-		return $retval
+	if [ $RETVAL -ne 0 ]; then
+		return $RETVAL
 	fi
 
-	echo -n $"Waiting for $prog to stop:"
+	show "Waiting for %s to stop" "$prog"
+	busy
 	timeout=10
 	while [ -e $runfile ]; do
 		sleep 1
 		timeout=$((timeout - 1))
 		if [ "$timeout" -le 0 ]; then
-			failure; echo
+			fail
 			return 1
 		fi
 	done
 
-	success; echo
+	ok
 	rm -f $lockfile
-	return $retval
+	return $RETVAL
 }
 
-restart() {
+condrestart() {
+	if [ ! -f /var/lock/subsys/sanlock ]; then
+		msg_not_running "sanlock"
+		RETVAL=$1
+		return
+	fi
 	stop && start
-	retval=$?
-	return $retval
-}
-
-reload() {
-	restart
-}
-
-rh_status() {
-	status $prog
+	RETVAL=$?
+	return $RETVAL
 }
 
-rh_status_q() {
-	rh_status >/dev/null 2>&1
+restart() {
+	stop && start
 }
 
 case "$1" in
 	start)
-		rh_status_q && exit 0
-		$1
+		start
 		;;
 	stop)
-		rh_status_q || exit 0
-		$1
+		stop
 		;;
 	restart)
-		$1
+		restart
 		;;
-	reload)
-		rh_status_q || exit 7
-		$1
+	try-restart)
+		condrestart 0
 		;;
 	force-reload)
-		force_reload
+		condrestart 7
 		;;
 	status)
-		rh_status
-		;;
-	condrestart|try-restart)
-		rh_status_q || exit 0
-		restart
+		status --pidfile $runfile sanlock
+		RETVAL=$?
 		;;
 	*)
-		echo $"Usage $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
-		exit 2
+		msg_usage "$0 {start|stop|restart|try-restart|force-reload|status}"
+		exit 3
 esac
-exit $?
-
+exit $RETVAL
--- sanlock-2.4/init.d/wdmd.orig	2012-07-23 16:24:38.000000000 +0200
+++ sanlock-2.4/init.d/wdmd	2012-09-02 21:14:08.491267019 +0200
@@ -3,7 +3,7 @@
 # wdmd - watchdog multiplexing daemon
 #
 # chkconfig: 2345 97 03
-# description: starts and stops sanlock daemon
+# description: starts and stops wdmd daemon
 #
 
 
@@ -32,83 +32,69 @@ WDMDOPTS="-G $WDMDGROUP"
 [ -f /etc/sysconfig/$prog ] && . /etc/sysconfig/$prog
 
 start() {
-	[ -x $exec ] || exit 5
-
-	if [ ! -d /var/run/$prog ]; then
-		mkdir -p /var/run/$prog
-		[ -x /sbin/restorecon ] && restorecon /var/run/$prog
-	fi
+	# Check if the service is already running?
+	if [ -f /var/lock/subsys/wdmd ]; then
+		msg_already_running "wdmd"
+		return
+ 	fi
 
 	if [ ! -c /dev/watchdog ]; then
-		echo -n $"Loading the softdog kernel module: "
+		show "Loading the softdog kernel module: "
 		modprobe softdog && udevadm settle
-		[ -c /dev/watchdog ] && success || failure
-		echo
+		[ -c /dev/watchdog ] && ok || fail
 	fi
 
-	echo -n $"Starting $prog: "
+	msg_starting "wdmd"
 	daemon $prog $WDMDOPTS
-	retval=$?
-	echo
-	[ $retval -eq 0 ] && touch $lockfile
-	return $retval
+	RETVAL=$?
+	[ $RETVAL -eq 0 ] && touch $lockfile
+	return $RETVAL
 }
 
 stop() {
-	echo -n $"Stopping $prog: "
-	killproc -p $runfile $prog -TERM
-	retval=$?
-	echo
-	[ $retval -eq 0 ] && rm -f $lockfile
-	return $retval
+	if [ ! -f /var/lock/subsys/wdmd ]; then
+		msg_not_running "wdmd"
+		return
+	fi
+	killproc -p $runfile $prog
+	RETVAL=$?
+	[ $RETVAL -eq 0 ] && rm -f $lockfile
+	return $RETVAL
 }
 
-restart() {
+condrestart() {
+	if [ ! -f /var/lock/subsys/wdmd ]; then
+		msg_not_running "wdmd"
+		RETVAL=$1
+		return
+	fi
 	stop
 	start
 }
-
-reload() {
-	restart
-}
-
-rh_status() {
-	status $prog
-}
-
-rh_status_q() {
-	rh_status >/dev/null 2>&1
-}
-
+ 
 case "$1" in
 	start)
-		rh_status_q && exit 0
-		$1
+		start
 		;;
 	stop)
-		rh_status_q || exit 0
-		$1
+		stop
 		;;
 	restart)
-		$1
+		stop
+		start
 		;;
-	reload)
-		rh_status_q || exit 7
-		$1
+	try-restart)
+		condrestart 0
 		;;
 	force-reload)
-		force_reload
+		condrestart 7
 		;;
 	status)
-		rh_status
-		;;
-	condrestart|try-restart)
-		rh_status_q || exit 0
-		restart
+		status --pidfile $runfile wdmd
 		;;
 	*)
-		echo $"Usage $0 {start|stop|status|restart|condrestart|try-restart|reload|force-reload}"
-		exit 2
+		msg_usage "$0 {start|stop|restart|try-restart|force-reload|status}"
+		exit 3
 esac
-exit $?
+exit $RETVAL
 
